### DISCUSSION TITLE
Only have public read permissions on collection items used by Page Builder (M2A) on a published page.

### DISCUSSION THREAD

**Discussion by louis-lau** [directus/directus#16465] [Archive]
Hi!

Say I have a `pages` and a `text_blocks` collection. There's a builder (M2A) on the pages, where you can add the text blocks and other blocks as well.

The permissions are set so that it's only possible to read published pages. But, it would seem that full read access to the `text_blocks` collection is needed as well. This includes text blocks on unpublished pages, which sounds like a problem to me.

I would only want to allow access to text blocks that are used in published pages. Is this possible?

**Comment by rijkvanzanten** [comment]
Relational paths can be used in the permissions checks, so you can configure read access to text_blocks where "connected page is published"

**Reply by louis-lau** [reply]
Ah, it's good to hear that this should be possible, as that sounds exactly like hwat I want. However, it doesn't show up. It does the other way around, but that doesn't really help haha. 

<img src="https://user-images.githubusercontent.com/1346804/202238581-23750c9a-3056-4245-aa26-6b5548706c1c.png"  width="300">
<hr>
<img src="https://user-images.githubusercontent.com/1346804/202238703-0d6cfc2d-17ea-4bfd-9723-0e16c4e00ace.png"  width="300">

**Reply by louis-lau** [reply]
Do I need to set it using a raw value or something? OR am I missing something else?

**Reply by bart** [reply]
@louis-lau Did you find any solution to this? Having exactly the same question now! Thanks :)

**Reply by louis-lau** [reply]
I don't think so, I didn't end up using directus.

**Reply by bart** [reply]
Ah okay, thanks for your super quick reply! @rijkvanzanten Do you maybe have a solution or further information on this? Thanks a lot in advance.

**Reply by domdus** [reply]
M2A is sort of a one way relation.
It's not possible to access the parent (page) fields on the `any` item side (blocks) for such a relational permission check.

You need to go for O2M or M2M to be able to do this.

If this is not an option, then the only way I could think of is:
- add a status field to the blocks collection
- apply the status=published permission check on the blocks collection

Now you may do not want to manage each and every blocks visibility/statuses.
- create a flow that triggers on a page (status) change
- get the m2a assigned items (ids of blocks) from that page
- perform an update (status) on all keys in the blocks collection.
So assigned blocks inheriting the status from parent page.
